packages/smithy-aws-core/src/smithy_aws_core/credentials_resolvers/credentials_resolver_chain.py

```diff
@@ -0,0 +1,54 @@
+from typing import Callable, List
+
+from smithy_aws_core.credentials_resolvers import EnvironmentCredentialsResolver
+from smithy_aws_core.identity import AWSCredentialsIdentity, AWSCredentialsResolver
+from smithy_core.aio.interfaces.identity import IdentityResolver
+from smithy_core.exceptions import SmithyIdentityException
+from smithy_core.interfaces.identity import IdentityProperties
+
+import os
+
+
+def _env_creds_available() -> bool:
+    return bool(os.getenv("AWS_ACCESS_KEY_ID")) and bool(
+        os.getenv("AWS_SECRET_ACCESS_KEY")
+    )
```

Comment:
nit; this is probably the more pythonic way to do this check.
```suggestion
    return (
        "AWS_ACCESS_KEY_ID" in os.environ
        and "AWS_SECRET_ACCESS_KEY" in os.environ
    )
```

Reply:
I was using this check for both set and not-empty.  Its probably not strictly necessary to check not empty though.  The current SEP does not specify either way....

Reply:
If the value exists in the environment, we should probably use it. If it doesn't work, I think that's better than ignoring it and potentially using unintended credentials. That provides the least risk and requires the end user to resolve the ambiguity before moving forward.